core/src/main/java/org/infinispan/remoting/inboundhandler/BaseBlockingRunnable.java

```diff
@@ -28,10 +28,18 @@ protected BaseBlockingRunnable(BasePerCacheInboundInvocationHandler handler, Cac
 
    @Override
    public void run() {
+      InvocationStatus invocationStatus = null;
       try {
-         response = beforeInvoke();
-         if (response == null) {
-            response = handler.invokePerform(command);
+         invocationStatus = beforeInvoke();
+         switch (invocationStatus) {
+            case CONTINUE:
+               response = handler.invokePerform(command);
+               break;
+            case CACHE_NOT_FOUND:
+               response = CacheNotFoundResponse.INSTANCE;
+               break;
+            case DEFERRED:
+               return;
```

Comment:
@belaban What I had in mind is preventing the duplication of ordering code (in JGroups and in application). Looking into `AsyncRequestHandler`, it would be cool if the message could be considered delivered (therefore, allowing next messages) only after `Response.send(...)` is invoked.